ddl/ddl_api.go

```diff
@@ -5367,108 +5368,89 @@ func buildPlacementSpecReplicasAndConstraint(rule *placement.RuleOp, replicas ui
 					break
 				}
 			}
-			newRule.Count = cnt
 
-			newRule.LabelConstraints, err = placement.CheckLabelConstraints(strings.Split(strings.TrimSpace(labels), ","))
+			labelConstraints, err := placement.CheckLabelConstraints(strings.Split(strings.TrimSpace(labels), ","))
 			if err != nil {
 				break
 			}
-			rules = append(rules, newRule)
+			rules = append(rules, &placement.Rule{
+				Count:            cnt,
+				LabelConstraints: labelConstraints,
+			})
 		}
-		rule.Count = ruleCnt
-
-		if rule.Count > 0 {
-			rules = append(rules, rule)
+		if ruleCnt > 0 {
+			rules = append(rules, &placement.Rule{
+				Count: ruleCnt,
+			})
 		}
 	} else {
 		err = errors.Errorf("constraint should be a JSON array or object, but got '%s'", cnstr)
 	}
 	return rules, err
 }
 
-func buildPlacementSpecs(specs []*ast.PlacementSpec) ([]*placement.RuleOp, error) {
-	rules := make([]*placement.RuleOp, 0, len(specs))
-
+func buildPlacementSpecs(bundle *placement.Bundle, specs []*ast.PlacementSpec) (*placement.Bundle, error) {
 	var err error
-	var sb strings.Builder
-	restoreFlags := format.RestoreStringSingleQuotes | format.RestoreKeyWordLowercase | format.RestoreNameBackQuotes
-	restoreCtx := format.NewRestoreCtx(restoreFlags, &sb)
+	var spec *ast.PlacementSpec
 
-	for _, spec := range specs {
-		rule := &placement.RuleOp{
-			Rule: &placement.Rule{
-				GroupID:  placement.RuleDefaultGroupID,
-				Override: true,
-			},
-		}
+	for _, rspec := range specs {
+		spec = rspec
 
+		var role placement.PeerRoleType
 		switch spec.Role {
 		case ast.PlacementRoleFollower:
-			rule.Role = placement.Follower
+			role = placement.Follower
 		case ast.PlacementRoleLeader:
-			rule.Role = placement.Leader
+			role = placement.Leader
 		case ast.PlacementRoleLearner:
-			rule.Role = placement.Learner
+			role = placement.Learner
 		case ast.PlacementRoleVoter:
-			rule.Role = placement.Voter
+			role = placement.Voter
 		default:
 			err = errors.Errorf("unknown role: %d", spec.Role)
+			break
 		}
 
-		if err == nil {
```

Comment:
Don't lose this judgement.

Reply:
So that is why I am using label break...